static/js/components/UserMenu.js

```diff
@@ -25,10 +25,41 @@ export const DropUpArrow = () => (
   <i className="material-icons arrow_drop_up">arrow_drop_up</i>
 )
 
+const loggedInMenuItems = () => {
+  return [
+    <Link key="1" to={SETTINGS_URL}>
+      Settings
+    </Link>,
+    SETTINGS.profile_ui_enabled && SETTINGS.username ? (
+      <Link key="2" to={profileURL(SETTINGS.username)}>
+        Profile
+      </Link>
+    ) : null,
+    SETTINGS.allow_email_auth ? (
+      <a key="3" href="/logout">
+        Sign Out
+      </a>
+    ) : null
+  ]
+}
+
+const loggedOutMenuItems = () => {
+  return [
+    <Link key="1" to={LOGIN_URL}>
+      Log In
+    </Link>,
+    <Link key="2" to={REGISTER_URL}>
+      Sign Up
+    </Link>
+  ]
```

Comment:
it looks to me like these two arrays of components could be declared either as:

1. constant arrays, since it doesn't look like there are any dependencies on props or other things that could change at runtime.
2. stateless react components with the series of links wrapped with `React.Fragment`, which has the advantage of avoiding the `key` prop. it would be good to also pull the `<li>` items out of the `UserMenu` below and wrap the links here.

simple react components feels a little more like idiomatic React to me, but up to you. It would let you do something down below like

```js
{showUserMenu ? (
  <DropdownMenu>
    {profile ? <LoggedInMenuItems /> : <LoggedOutMenuItems />}
  </DropdownMenu>
) : null }
```

which avoids the need for mapping across the menuitems and so on down below.

Reply:
also, should we add some basic assertions to the `UserMenu` tests about when one or the other list of links is used, and what is included?

Reply:
Great catch. I meant to include a test for the menu contents but it slipped my mind. I'll also look at your first comment. I decided on an array of `<li>` since the wrapping `<ul>` should be the same between both.